src/tests/class_tests/openms/source/TargetedSpectraExtractor_test.cpp

```diff
@@ -0,0 +1,698 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2017.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/test_config.h>
+
+///////////////////////////
+#include <OpenMS/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.h>
+///////////////////////////
+
+using namespace OpenMS;
+using namespace std;
+
+START_TEST(TargetedSpectraExtractor, "$Id$")
+
+/////////////////////////////////////////////////////////////
+// Raw spectrum data acquired in DDA mode (i.e., product ion full spectrum scan)
+// measured on a QTRAP 5500 corresponding to C-Aconitate
+// taken from E. coli grown on glucose M9 during steady-state
+// for flux analysis.
+
+MSSpectrum spectrum;
+spectrum.resize(73);
+MSSpectrum::Iterator it = spectrum.begin();
+
+it->setMZ(61.92);
+it++->setIntensity(6705.41660838088f);
```

Comment:
looks a bit strange to my eyes. Other tests just load a dta file.
If you want to have these values in a human readable way in the test you could just use intitializer lilsts
and add the peaks in a loop. 
e.g.:
```
  std::vector<int> mz = {1,2,3}; 
  std::vector<int> intensity= {1,2,3}; 
  for (Size i = 0; i != mz.size(); ++i) { spectrum.push_back(Peak1D(mz[i],intensity[i])); }
```

Would be shorter and imho a bit easier to read.

Reply:
Thanks, I modified it and I also made sure that I use `Size` instead of `UInt` when dealing with `.size()` return values.